### DISCUSSION TITLE
VirtWifi feature for certain apps or games.

### DISCUSSION THREAD

**Discussion by jm2k7** [waydroid/waydroid#1593] [Ideas]
Hi guys, I'm having issues with "fake wifi" not working in certain apps, I find it ironic that native Android-x86 and other emulators that run on custom virtual machines in Windows already have that feature.

An example of how virtwifi works efficiently "android-x86_64-9.0-r2":
![Ubuntu 64-bit-2024-10-15-06-59-07](https://github.com/user-attachments/assets/9c0499bb-92c8-4221-b053-7d4e1890f9a9)
![Ubuntu 64-bit-2024-10-15-06-59-20](https://github.com/user-attachments/assets/9fa092f1-8026-465d-a793-6d95b659f6b8)
![Ubuntu 64-bit-2024-10-15-07-26-14](https://github.com/user-attachments/assets/fa444f59-5b3d-481a-9571-913194daebaa)
![Ubuntu 64-bit-2024-10-15-07-30-05](https://github.com/user-attachments/assets/4afe7c96-eba4-4be4-b39f-19f8541537ee)

Anyway, it is not mandatory, but if someone can implement this feature, it would be appreciated in advance.

Greetings.

**Comment by baptx** [comment]
For me also, an Android app was only working using Android-x86 on virt-manager.
For Waydroid, I had to use a workaround to bypass the network check (I don't know if the workaround was needed for Android 11 but fake_wifi may not work for some apps in Android 13):
https://github.com/waydroid/waydroid/issues/619#issuecomment-4061247917
